taxonomy/api.py

```diff
@@ -1,252 +1,54 @@
 """APIs for lore taxonomy application"""
 from __future__ import unicode_literals
 
-from django.db import transaction
-from django.shortcuts import get_object_or_404
-
 from taxonomy.models import (
     Vocabulary,
     Term,
 )
-from learningresources.models import (
-    LearningResource,
-    LearningResourceType,
+from learningresources.api import (
+    get_repo,
+    NotFound,
 )
 
 
-def get_vocabulary(vocabulary_id):
+def get_vocabulary(repo_slug, user_id, vocab_slug):
     """
-    Lookup vocabulary given its id.
+    Lookup vocabulary given its slug, using repo_slug to validate ownership.
 
     Args:
-        vocabulary_id (int): Vocabulary id
+        repo_id (int): Repository id
+        user_id (int): User id
+        vocab_slug (unicode): Vocabulary slug
 
     Returns:
         Vocabulary (Vocabulary): The vocabulary from the database
     """
-    return get_object_or_404(Vocabulary, id=vocabulary_id)
+    repo = get_repo(repo_slug, user_id)
```

Comment:
This could break if the slug is bad. It should be possible to put it in the `try/except` below and replace `Vocabulary.DoesNotExist` with `models.DoesNotExist`.